src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -937,6 +967,77 @@ impl Edges {
 
 
 
+// =========================
+// === ExecutionStatuses ===
+// =========================
+
+/// [`ExecutionsStatuses`] can be used to collect the execution statuses of all nodes in a graph. It
+/// exposes their minimum and maximum running times through its FRP endpoints. The structure needs
+/// to be updated whenever a node is added or deleted or changes its execution status.
+mod execution_statuses {
+    use super::*;
+    use std::cmp::Ordering::Equal;
+
+    ensogl::define_endpoints! {
+        Output {
+            min_duration (f32),
+            max_duration (f32),
+        }
+    }
+
+    #[derive(Debug,Clone,CloneRef,Default)]
+    pub struct ExecutionStatuses {
+        frp       : Frp,
+        // If performance becomes an issue then we could try to store the durations in efficient
+        // priority queues instead, one to determine the minimum and one to determine the maximum.
```

Comment:
We could also use https://docs.rs/bimap/0.6.1/bimap/btree/struct.BiBTreeMap.html (I already used bimap crate somewhere else in the project).

Reply:
If we want to avoid additional dependencies then I would rather use `BTreeSet` that seems more convenient and more efficient at first sight.

Reply:
Do you agree?

Reply:
It is already a dependency. And why BTreeSet is more convenient? the BiBTreeMap will replace both the "direct" map nodeId->time and the set of times, and will synchronize them. And it should not be inefficient as it uses the BTreeSet itself.

Reply:
Thinking about it again, `BTreeSet` would not work of course, because we lose the connection to the nodes. But I could not find any reasonable way to access the maximum through the interface of `BiBTreeMap`. Did I miss something?

Reply:
You can use the https://docs.rs/bimap/0.6.1/bimap/btree/struct.BiBTreeMap.html#method.right_values iterator - it is in ascending order and double ended, so you can quickly get the first and last element.

Reply:
Thanks! I did not notice that it was double ended.